src/hooks/useFunctionIds.ts

```diff
@@ -1,32 +1,32 @@
-/*
- * Copyright 2020 RoadieHQ
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-import { useAsync } from 'react-use';
-import { useSettings } from './useSettings';
-
-export const FIREBASE_FUNCTION_IDS = 'cloud.google.com/function-ids';
-
-export const useFunctionIds = () => {
-  let [{ entity }] = useSettings();
-
-  const { value, loading, error } = useAsync(async () => {
-    const rawProjects =
-      entity?.metadata.annotations?.[FIREBASE_FUNCTION_IDS] ?? '';
-    const functions = rawProjects.split(',').map(p => p.trim());
-    const availableProjects = [...new Set(functions.map(f => f.split('/')[1]))];
-    return { availableProjects, functions };
-  });
-  return { value, loading, error };
-};
+/*
+ * Copyright 2020 RoadieHQ
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { useAsync } from 'react-use';
+import { useSettings } from './useSettings';
+
+export const FIREBASE_FUNCTION_IDS = 'cloud.google.com/function-ids';
+
+export const useFunctionIds = () => {
+  const [{ entity }] = useSettings();
+
+  const { value, loading, error } = useAsync(async () => {
+    const rawProjects =
+      entity?.metadata.annotations?.[FIREBASE_FUNCTION_IDS] ?? '';
+    const functions = rawProjects.split(',').map(p => p.trim());
+    const availableProjects = [...new Set(functions.map(f => f.split('/')[1]))];
+    return { availableProjects, functions };
+  });
+  return { value, loading, error };
+};
```

Comment:
```suggestion
    const rawProjects =
      entity?.metadata.annotations?.[FIREBASE_FUNCTION_IDS] ?? '';
    const functions = rawProjects.split(',').map(p => p.trim());
    const availableProjects = [...new Set(functions.map(f => f.split('/')[1]))];
  return { value: { availableProjects, functions }, loading: false, error: false };
};
```

Reply:
Since entity is available without the need for request to api, we dont need useAsync anymore

Reply:
fixed